rfcs/convergence/popper-js.md

```diff
@@ -0,0 +1,136 @@
+This section is important as the motivation or problem statement is indepenent from the proposed change. Even if this RFC is not accepted this Motivation can be used for alternative solutions.
+
+In the end, please make sure to present a neutral Problem statement, rather than one that motivates a particular solution
+-->
+
+Positioning is a hard problem to solve. Pure CSS solutions are not powerful enough in modern applications since they have
+accessibility and usability problems:
+
+- CSS tooltips will not take into account boundaries
+- CSS will not flip and reposition based on available viewport space
+- Cannot follow mouse cursor or be used as a context menu
+- Hard to modify the behaviour at runtime
+
+Therefore, most solutions are writting in complex javascript that requires handling of all the boundary and scroll cases
+at runtime. This results in very complicated code, that needs to be actively maintained and improved. Looking
+[at the current v7 documentation of Callout](https://developer.microsoft.com/en-us/fluentui#/controls/web/callout), there are
+more than 20 issues that stretch back to 2018 that still have not been resolved.
+
+Popper references and supports the following selection of v8 open issues (following the thread discussions):
+
+- [#7970](https://github.com/microsoft/fluentui/issues/7970) - Beak positioning and offset
+- [#9249](https://github.com/microsoft/fluentui/issues/9249) - Modifiers to create sane resizing/repositioning variants of popper
+- [#5148](https://github.com/microsoft/fluentui/issues/5148) - Decoupled placement properties for each access for easy RTL modifiers
+
+Convergence is moving fast, spending time to build a fully-fledged positioning library that handles all the boundary cases
+can be time consuming and slow down or block development for some components
+
+## Detailed Design or Proposal
+
+<!-- This is the bulk of the RFC. Explain the proposal or design in enough detail for the inteded audience to understand. -->
+
+The important part of this RFC is to get quorum on using popper in Fluent. This will allow us to offload the heavy effort
+of managing out of order DOM popoverse and focus on delivering value to our partners through usable components.
+
+As mentioned above, out of the box popper provides support and configuration for the following scenarios:
+
+- Preventing overflow on available space and window resize
+- Content repositioning on window scroll
+- Content flipping based on available space
+- Content offset from anchor element
+- Beak/Arrow positioning
+
+In the short/long-term future, it is always possible for the team to reevaluate the scenarios supported by popper to see
+if the the library needs to be extended through [popper modifiers](https://popper.js.org/docs/v2/modifiers/) or a custom
+solution, if there is a need for it based on feedback from our partners.
+
+This RFC does not specify exactly how popper should be integrated into fluent, and is left to a technical discussion or
```

Comment:
Hmm, if there are no proposed guidelines for using it, shouldn't this be a proposal issue rather than an RFC?

Reply:
I guess you have seen through my OSS warrior disguise 😆 

That's a fair point, I don't know how things are proposed for convergence currently, would be worth bringing up at the next sync

Reply:
WDYT about RFC/proposal issues @JustSlone  ?

Reply:
So my proposal for this would be that questions/discussion on topics like this one could easily go as actual issues in the repo (with the entire description that you have masterfully crafted as the contents of this RFC), and leave the RFC proper for feature/design decisions regarding the converged components. I'm curious what y'all think about that!

Reply:
Maybe I'm odd, but I actually think this particular RFC is a good use of the process. Perhaps it is not complete with a full proposal for implementation, but it raises a idea, looks for feedback, and we can leverage the Merge/Close functionality to record the outcome.

Yeah we might find once we dig into implementation details that popper is a bad idea, but that's OK, we can always reverse this later. 

It is true that we could use an issue could capture this in a lighter weight manner, and it is very much true that here the line-by-line commenting PRs give us is not particularly needed. 
However:
- issues more easily drift off into the ether, due to not having clear Merge/Close states
- it seems easier to hold our discussions in one place (especially as we are getting used to async discussions), rather than spread across both PRs and issues

My thoughts here could be summarized as, RFC/proposal Issues are probably a good idea eventually, but let's get comfortable with our PR RFC process first. 

WDYT? Is there something problematic with doing this RFC/proposal as a PR?

Reply:
Adding another perspective, the goal is that we figure out if we should / should not use popper and the real goal is that we have components that can "popup." Let's be careful of the meta work, like formalizing posts.

That said, I would say an RFC is good use here.